src/graphql/data/nft/NftUniversalRouterAddress.ts

```diff
@@ -0,0 +1,26 @@
+import gql from 'graphql-tag'
+import { useMemo } from 'react'
+
+import { useNftUniversalRouterAddressQuery } from '../__generated__/types-and-hooks'
+
+gql`
+  query NftUniversalRouterAddress($chain: Chain = ETHEREUM) {
+    nftRoute(chain: $chain, senderAddress: "", nftTrades: []) {
+      toAddress
```

Comment:
Wondering if we considered avoiding the extra call by getting `toAddress` to the `nftRoute` call we make regardless during purchase and kicking off the approval process after that, based on that.

Reply:
they thought it should be approve -> pay isntead of pay -> approve -> pay